package/MDAnalysis/visualization/streamlines_3D.py

```diff
@@ -64,16 +74,34 @@ def determine_container_limits(coordinate_file_path, trajectory_file_path, buffe
 
 
 def produce_grid(tuple_of_limits, grid_spacing):
-    '''Produce a grid for the simulation system based on the tuple of Cartesian Coordinate limits calculated in an
-    earlier step.'''
+    """Produce a grid for the simulation system.
+
+    The partitioning is based on the tuple of Cartesian Coordinate limits
+    calculated in an earlier step.
+    """
     x_min, x_max, y_min, y_max, z_min, z_max = tuple_of_limits
     grid = np.mgrid[x_min:x_max:grid_spacing, y_min:y_max:grid_spacing, z_min:z_max:grid_spacing]
     return grid
 
 
 def split_grid(grid, num_cores):
-    '''Take the overall grid for the system and split it into lists of cube vertices that can be distributed to each
-    core.'''
+    """Split the grid into blocks of vertices.
+
+    Take the overall `grid` for the system and split it into lists of cube
+    vertices that can be distributed to each core.
+
+    Parameters
+    ----------
+    grid : numpy.array
+        3D array
+    num_cores : int
+        number of partitions to generate
+
+    Returns
+    -------
+    [list_dictionaries_for_cores, total_cubes, num_sheets, delta_array_shape]
+
+    """
```

Comment:
refactor into underscore-quoted helpers would be good but without tests I am not doing it – I leave this to you :-)